Tests/SonarScanner.MSBuild.PreProcessor.Test/WebClientDownloaderTest.cs

```diff
@@ -144,9 +154,102 @@ public void MultipleDisposeCallsNotFailing()
             testDownloader.IsDisposedCalled.Should().BeTrue();
         }
 
+        [TestMethod]
+        public async Task DownloadStream_Success()
+        {
+            var logger = new TestLogger();
+            var sut = CreateSut(logger, HttpStatusCode.OK);
+
+            var stream = await sut.DownloadStream(testUri);
+
+            var text = await new StreamReader(stream).ReadToEndAsync();
+            text.Should().Be(TestContent);
+            logger.AssertDebugLogged("Downloading from https://www.sonarsource.com/...");
+        }
+
+        [TestMethod]
+        public async Task DownloadStream_Fail()
+        {
+            var logger = new TestLogger();
+            var sut = CreateSut(logger, HttpStatusCode.NotFound);
+
+            var stream = await sut.DownloadStream(testUri);
+
+            stream.Should().BeNull();
+            logger.AssertDebugLogged("Downloading from https://www.sonarsource.com/...");
+            logger.AssertNoWarningsLogged();
```

Comment:
Isn't this behavior weird? Or at least not helpful?

If it fails, I'd expect INFO or at least DEBUG message saying that server returned 404 not found. Or something like that. Actually for `Download` as well as `DownloadStream`

Reply:
Indeed. I've preserved the original behavior but it doesn't make sense. I'll add some logging.